### PR TITLE
Add start up code to worker

### PR DESCRIPTION
**Pull request by maiqbal11**
Fixes https://github.com/Azure/azure-functions-python-worker/issues/547

### CODE REVIEW

File: azure_functions_worker/constants.py [hunk 1]
```diff
@@ -1,3 +1,10 @@
 # Capabilities
 RAW_HTTP_BODY_BYTES = "RawHttpBodyBytes"
 TYPED_DATA_COLLECTION = "TypedDataCollection"
+
+# User packages
+PKGS_PATH = "site/wwwroot/.python_packages"
+VENV_PKGS_PATH = "site/wwwroot/worker_venv"
+
+PKGS_36 = "lib/python3.6/site-packages"
+PKGS_37 = "lib/python3.7/site-packages"
```
Comment (Hazhzeng) [inline azure_functions_worker/constants.py]: In PGKS_PATH, we want to drop the version specific package path.
We can add a python path to support **lib/site-packages** as well, so that we will have a transition windows for updating our toolings.

File: azure_functions_worker/main.py [hunk 2]
```diff
@@ -26,8 +30,28 @@ def parse_args():
...
+    home = Path.home()
+    pkgs_path = os.path.join(home, constants.PKGS_PATH)
+    venv_pkgs_path = os.path.join(home, constants.VENV_PKGS_PATH)
+
+    if minor_version == 6:
+        sys.path.insert(1, os.path.join(venv_pkgs_path, constants.PKGS_36))
+        sys.path.insert(1, os.path.join(pkgs_path, constants.PKGS_36))
+    elif minor_version == 7:
+        sys.path.insert(1, os.path.join(venv_pkgs_path, constants.PKGS_37))
+        sys.path.insert(1, os.path.join(pkgs_path, constants.PKGS_37))
```
Comment (ankitkumarr) [inline azure_functions_worker/main.py]: out of curiosity, why 1 and not 0 for these?

**Comment by ankitkumarr** [comment]
With this change, we would update the path even if we are running the worker locally. Is that expected? If there's a way to tell whether or not we are running it in Azure context, maybe we can use that for updating the path.

Probably not a big deal though, as invalid entries in path are usually harmless.

**Comment by Hazhzeng** [comment]
Thanks @maiqbal11 for reminding me of this.

Python 3.6 Paths:
- site/wwwroot/.python_packages/lib/python3.6/site-packages
- site/wwwroot/.python_packages/lib/site-packages
- site/wwwroot/worker_venv/lib/python3.6/site-packages
- site/wwwroot/worker_venv/lib/site-packages

Python 3.7 Paths:
- site/wwwroot/.python_packages/lib/python3.7/site-packages
- site/wwwroot/.python_packages/lib/site-packages
- site/wwwroot/worker_venv/lib/python3.7/site-packages
- site/wwwroot/worker_venv/lib/site-packages

File: azure_functions_worker/main.py (hunk 2 above)
Comment (Hazhzeng) [reply azure_functions_worker/main.py]: We may want the currently directory path to have the top priority.

Comment (maiqbal11) [reply azure_functions_worker/main.py]: I had wanted to keep the current directory path (i.e. `''`) as the first search path. This mimics the current behavior of the `PYTHONPATH` which is appended as the second entry in the `sys.path` after the current directory.

**Comment by ankitkumarr** [comment]
@Hazhzeng, thanks!

I think we only need --

Python 3.6 Paths:
- site/wwwroot/.python_packages/lib/python3.6/site-packages
- site/wwwroot/.python_packages/lib/site-packages
- site/wwwroot/worker_venv/lib/python3.6/site-packages

Python 3.7 Paths:
- site/wwwroot/.python_packages/lib/site-packages

Hoping that no point of time would we need to deploy a virtual environment.
Is there a reason you included the others?

**Comment by Hazhzeng** [comment]
> @Hazhzeng, thanks!
> 
> I think we only need --
> 
> Python 3.6 Paths:
> 
> * site/wwwroot/.python_packages/lib/python3.6/site-packages
> * site/wwwroot/.python_packages/lib/site-packages
> * site/wwwroot/worker_venv/lib/python3.6/site-packages
> 
> Python 3.7 Paths:
> 
> * site/wwwroot/.python_packages/lib/site-packages
> 
> Adding, `site/wwwroot/.python_packages/lib/python3.7/site-packages` as a safety net for the future is fine. But not sure if we need others.
> Hoping that no point of time would we need to deploy a virtual environment.
> Is there a reason you included the others?

One of my concern for **worker_venv** is in the Azure DevOps Python deployment page https://docs.microsoft.com/en-us/azure/azure-functions/functions-how-to-azure-devops#python.

Currently, we ask our customers to resolve the dependencies by running **python3.6 -m venv worker_venv**.

Desperately, the virtual environment path is different depends on the OS on the devops agent. 
On windows: worker_venv\Lib\site-packages
On ubuntu: worker_venv/lib/python3.6/site-packages

In conclusion, we only need to support `site/wwwroot/.python_packages/lib/site-packages` path for Python 3.7 and future versions.

**Comment by maiqbal11** [comment]
@ankitkumarr, @Hazhzeng, please take look. I believe all comments should be addressed.

**Comment by maiqbal11** [comment]
Closing this in favor of consolidating all changes under https://github.com/Azure/azure-functions-python-worker/pull/555. It makes sense of the generated NuGet to contain these worker changes since they are essential for the new NuGet to work. I've created a separate commit for the changes in this PR here: https://github.com/Azure/azure-functions-python-worker/pull/555/commits/107d6952bca9ea1ef88201e99f781124ad87c20a. I needed to included all the startup code in `worker.py` because adding to `sys.path` directly was not registering user packages. We need to add them to the `PYTHONPATH` before we spin up a new python process with the worker.
